### PR TITLE
[feature][connector-v2][elasticsearch] Support write cdc changelog event in elasticsearch sink

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

Support write cdc changelog event in elasticsearch sink

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by hailin0** [comment]
@EricJoy2048 PTAL

File: docs/en/connector-v2/sink/Elasticsearch.md [hunk 1]
```diff
@@ -58,10 +66,28 @@ batch bulk doc max size
...
+
+CDC(Change data capture) event
+
+```bash
+sink {
+    Elasticsearch {
+        hosts = ["localhost:9200"]
+        index = "seatunnel-${age}"
+        
+        primary_keys = ["key1", "key2", ...]
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/Elasticsearch.md:91]: I think `primary_keys ` is required when use cdc, right？

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 2]
```diff
@@ -45,15 +47,63 @@ public class ElasticsearchRowSerializer implements SeaTunnelRowSerializer {
...
+    }
+
+    private String serializeUpsert(SeaTunnelRow row) {
+        String key = keyExtractor.apply(row);
+        Map<String, Object> document = toDocumentMap(row);
+
+        try {
+            if (key != null) {
+                Map<String, String> upsertMetadata = createMetadata(row, key);
+                return String.format("{ \"update\" : %s }\n{ \"doc\" : %s, \"doc_as_upsert\" : true }",
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java]: `serializeUpsert ` will be called by every row. So, I suggest we use better way to process String.

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 3]
```diff
@@ -66,25 +116,20 @@ public String serializeRow(SeaTunnelRow row) {
...
+    private Map<String, String> createMetadata(@NonNull SeaTunnelRow row,
+                                               @NonNull String key) {
+        Map<String, String> actionMetadata = createMetadata(row);
+        actionMetadata.put("_id", key);
+        return actionMetadata;
+    }
 
-        return sb.toString();
+    private Map<String, String> createMetadata(@NonNull SeaTunnelRow row) {
+        Map<String, String> actionMetadata = new HashMap<>();
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java]: gave a init capacity？

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/KeyExtractor.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    public static Function<SeaTunnelRow, String> createKeyExtractor(SeaTunnelRowType rowType,
+                                                                    String[] primaryKeys,
+                                                                    String keyDelimiter) {
+        if (primaryKeys == null) {
+            return row -> null;
+        }
+
+        List<FieldFormatter> fieldFormatters = new ArrayList<>();
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/KeyExtractor.java]: gave an init capacity?

**Comment by hailin0** [comment]
@iture123 PTAL

**Review by EricJoy2048 — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**